src/pages/Details.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+
+export default function Details() {
+  const currencies = useSelector((state) => state.currency.allData);
+  const { currency } = useParams();
+  const [currencyValues, setCurrencyValues] = React.useState({});
+
+  function getKeyByValue(object, value) {
+    return Object.keys(object).find((key) => object[key] === value);
+  }
+
+  const key = getKeyByValue(currencies, currency);
+  const keyUpper = key && key.toUpperCase();
+
+  const date = new Date().toISOString().slice(0, 10);
+  React.useEffect(() => {
+    fetch(`https://cdn.jsdelivr.net/gh/fawazahmed0/currency-api@1/${date}/currencies/usd.json`)
+      .then((response) => response.json())
+      .then((data) => {
+        setCurrencyValues(data.usd);
+      });
+  }, [date]);
+
```

Comment:
**OPTIONAL** - Good job so far, but it would have been neater if this API call is made in your redux store. You can consider creating a `DetailsSlice` and dispatch the API call in your `useEffect` hook. Please fix this 🙏

Reply:
Thank you, Sir. I will surely implement that, I just wanted to practice every different approach I have learned so far.